### PR TITLE
Use ChefConfig::Config.from_file to load config files

### PR DESCRIPTION
**Pull request by mcquin**
Resolves https://github.com/chef-cookbooks/chef-client/issues/387

\cc @tas50

### CODE REVIEW

**Comment by tas50** [comment]
@mcquin Will this work with Chef 11 releases?

**Comment by mcquin** [comment]
Oh, probably not. Let me see what I can do about that.

**Comment by tas50** [comment]
+1

**Comment by mcquin** [comment]
moar review plz @thommay @iennae :)

File: templates/default/client.rb.erb [hunk 1]
```diff
@@ -69,5 +69,9 @@ rescue NameError => e
 end
 
 Dir.glob(File.join(<%= node['chef_client']['conf_dir'].inspect %>, "client.d", "*.rb")).each do |conf|
-  Chef::Config.from_file(conf)
+  <% if Gem::Version.new(Chef::VERSION) < Gem::Version.new('12.4.0') -%>
+    Chef::Config.from_file(conf)
```
Comment (tas50) [inline templates/default/client.rb.erb]: Can we avoid the indents here even though it makes the ERB harder to read

Comment (mcquin) [inline templates/default/client.rb.erb]: You don't like the 4-space indent? :P

**Pull request merged**